taipy/gui/servers/fastapi/server.py

```diff
@@ -0,0 +1,378 @@
+
+    def _get_default_handler(
+        self,
+        static_folder: str,
+        template_folder: str,
+        title: str,
+        favicon: str,
+        root_margin: str,
+        scripts: t.List[str],
+        styles: t.List[str],
+        version: str,
+        client_config: t.Dict[str, t.Any],
+        watermark: t.Optional[str],
+        css_vars: str,
+        base_url: str,
+    ) -> APIRouter:
+        templates = Jinja2Templates(directory=template_folder)
+
+        taipy_router = APIRouter()
+
+        async def emit_dispatcher():
+            while True:
+                emit_args, emit_kwargs = await self._emit_queue.get()
+                try:
+                    await self._ws.emit(*emit_args, **emit_kwargs)
+                except Exception as e:
+                    _TaipyLogger._get_logger().info("Error sending ws message: ", e)
+
+        @asynccontextmanager
+        async def lifespan(app: FastAPI):
+            self._emit_queue = Queue()
+            self._ws.start_background_task(emit_dispatcher)
+            yield
+
+        taipy_router.lifespan_context = lifespan
+
+        @taipy_router.get("/", response_class=HTMLResponse)
+        @taipy_router.get("/{path:path}", response_class=HTMLResponse)
+        def my_index(request: Request, path: str = "", request_meta: _AppCtxGlobals = Depends(request_meta_dependency)):  # noqa: B008
+            with set_request_ctx(request), get_request_meta_ctx():
+                resource_handler_id = request.cookies.get("_RESOURCE_HANDLER_ARG", None)
+                if resource_handler_id is not None:
+                    resource_handler = _ExternalResourceHandlerManager().get(resource_handler_id)
+                    if resource_handler is None:
+                        reload_html = """
+                            <html>
+                                <head><style>body {background-color: black; margin: 0;}</style></head>
+                                <body><script>location.reload();</script></body>
+                            </html>
+                        """
+                        response = HTMLResponse(content=reload_html, status_code=400)
+                        response.set_cookie(
+                            "_RESOURCE_HANDLER_ARG",
+                            "",
+                            secure=request.url.scheme == "https",
+                            httponly=True,
+                            expires=0,
+                            path="/",
+                        )
+                        return response
+                    try:
+                        return resource_handler.get_resources(path, static_folder, base_url)
+                    except Exception as e:
+                        raise HTTPException(
+                            status_code=500, detail="Can't get resources from custom resource handler"
+                        ) from e
+
+                if not path or path == "index.html" or "." not in path:
```

Comment:
I'm not sure I understand the '"." not in path" part.
Please explain (then comment),